src/test/rustdoc/extern-default-method.rs

```diff
@@ -1,9 +1,21 @@
 // aux-build:rustdoc-extern-default-method.rs
 // ignore-cross-compile
+// ignore-tidy-linelength
 
 extern crate rustdoc_extern_default_method as ext;
 
+// We didn't built the documentation of the dependency. The struct and its impl from the
+// external crate should still be documented since it is re-exported from the this crate.
+// However, the method in the trait impl should *not* have a link (an `href` attribute) to
+// its corresponding item in the trait declaration since it would be broken.
+//
+// In older versions of rustdoc, the impl item (`a[@class="fnname"]`) would link to
+// `#method.provided` – i.e. "to itself". Put in quotes since that was actually incorrect in
+// general: If the type `Struct` also had an inherent method called `provided`, the impl item
+// would link to that one even though those two methods are distinct items!
+
 // @count extern_default_method/struct.Struct.html '//*[@id="method.provided"]' 1
-// @has extern_default_method/struct.Struct.html '//*[@id="method.provided"]//a[@class="fnname"]/@href' #method.provided
+// @!has extern_default_method/struct.Struct.html '//*[@id="method.provided"]//a[@class="fnname"]/@href'
```

Comment:
`@!has` is okay / robust here since it's immediately followed by a `@count … 1` check. I want to check that the element exists but the attribute `href` on it doesn't.

Reply:
We tend to prefer `@snapshot` over `@!has`. What do you think?

Reply:
Or instead of checking what isn't there, check what's actually there in addition to the `@count`?

Reply:
Okay, I will use `@snapshot` then.